### ISSUE TITLE
Uncaught Error: ENOENT, no such file or directory '/Users/davidhooey/.atom/packages/autocomplete-plus/lib/provider.coffee'

### ISSUE THREAD

**Issue by davidhooey** [atom/settings-view#388]
[Enter steps to reproduce below:]

1. Go to Settings -> Updates and select Update All.

**Atom Version**: 0.182.0
**System**: Mac OS X 10.10.2
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.183.0

### Stack Trace

Uncaught Error: ENOENT, no such file or directory '/Users/davidhooey/.atom/packages/autocomplete-plus/lib/provider.coffee'

```
At /Applications/Atom.app/Contents/Resources/app/src/package.js:241

Error: ENOENT, no such file or directory '/Users/davidhooey/.atom/packages/autocomplete-plus/lib/provider.coffee'
  at Error (native)
  at Object.fs.openSync (fs.js:503:18)
  at Object.module.(anonymous function) [as openSync] (/Applications/Atom.app/Contents/Resources/atom/common/lib/asar.js:422:20)
  at Object.fs.readFileSync (fs.js:355:15)
  at Object.fs.readFileSync (/Applications/Atom.app/Contents/Resources/atom/common/lib/asar.js:329:27)
  at Object.exports._compileFile (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:206:14)
  at getSourceMap (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:314:22)
  at getSourceMapping (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:322:19)
  at formatSourcePosition (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:274:16)
  at /Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:340:34
  at Function.Error.prepareStackTrace (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/coffee-script.js:343:7)
  at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app/src/package.js:241:79)
  at /Applications/Atom.app/Contents/Resources/app/src/package.js:221:28
  at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:167:15)
  at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app/src/package.js:215:14)
  at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:464:21)
  at /Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/package-manager.js:226:29
  at exit (/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/package-manager.js:43:16)
  at BufferedProcess.triggerExitCallback (/Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:63:47)
  at /Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:77:18
  at Socket.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:150:18)
  at Socket.emit (events.js:138:20)
  at Pipe.close (net.js:477:12)

```

### Commands

```
     -2:11.0 command-palette:toggle (atom-text-editor.editor)
     -2:09.8 core:backspace (atom-text-editor.editor.mini)
  2x -2:06.3 core:move-down (atom-text-editor.editor.mini)
  2x -2:05.5 core:move-up (atom-text-editor.editor.mini)
  4x -2:04.3 core:move-down (atom-text-editor.editor.mini)
     -2:03.2 core:confirm (atom-text-editor.editor.mini)
     -2:03.2 settings-view:change-themes (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "themes": [
      "seti-ui",
      "railscast-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
emmet, v2.3.3
git-log, v0.3.0
language-haml, v0.15.0
less-autocompile, v0.3.3
linter, v0.11.1
linter-coffeelint, v0.1.7
linter-csslint, v0.0.11
linter-erb, v0.1.0
linter-htmlhint, v0.0.8
linter-javac, v0.1.3
linter-js-yaml, v1.0.0
linter-jshint, v0.1.0
linter-less, v0.3.1
linter-puppet-lint, v0.2.8
linter-pylint, v0.2.0
linter-ruby, v0.1.4
linter-scss-lint, v0.0.14
linter-write-good, v0.2.2
minimap, v4.3.1
railscast-theme, v0.3.0
seti-ui, v0.6.3

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
In all four of these cases the specified package does not actually seem to be installed.

Anyone have any idea what's going on here?

**Comment by matiasow** [comment]
No clue. But restarted the app (Atom > Quit and then open it again) and it worked now.

**Comment by 50Wliu** [comment]
@thedaniel @kevinsawicki @benogle @nathansobo @izuzak mind taking a look at this?

**Comment by benogle** [comment]
Wonder if this has to do with services?

**Comment by kevinsawicki** [comment]
This is really weird, does that file exist on disk?

**Comment by abe33** [comment]
Actually I think it's because it's the previous code of the package that get activated again after an update, and not the new code.

We had an issue of this kind in the minimap, see atom-minimap/minimap#274.

I got one recently when moving the minimap plugin generation view to use a custom element. The file was renamed, and the require being done lazily in the command, when I tried running the command after having updated the package I got the same kind of issue: It was still running the previous code referencing the old file name which no longer existed, leading to a ENOENT.
